contributors/design-proposals/architecture/scope.md

```diff
@@ -0,0 +1,342 @@
+  agree to maintain the functionality?
+
+* Do the contributors to the functionality agree to follow the
+  project’s development conventions and requirements, including CLA,
+  code of conduct, github and build tooling, testing, documentation,
+  and release criteria, etc.?
+
+* Does the functionality improve existing use cases, or mostly enable
+  new ones? The project isn't completely blocking new functionality
+  (more reducing the rate of expansion), but it is trying to
+  limit additions to kubernetes/kubernetes/master, and aims to improve the
+  quality of the functionality that already exists.
+
+* Is it needed by project contributors? Example: We need cluster
+  creation and upgrade functionality in order to run end-to-end tests.
+
+* Is it necessary in order to enable workload portability?
+
+* Is it needed in order for upstream releases to be usable? For
+  example, things without which users otherwise were
+  reverse-engineering Kubernetes to figure out, and/or copying code
+  out of Kubernetes itself to make work.
+
+* Is it functionality that users expect, such as because other
+  container platforms and/or service discovery and routing mechanisms
+  provide it? If a capability that relates to Kubernetes's fundamental
+  purpose were to become table stakes in the industry, Kubernetes
+  would need to support it in order to stay relevant. (Whether it
+  would need to be addressed by the core project would depend on the
+  other criteria.)
+
+* Is there sufficiently broad user demand and/or sufficient expected
+  user benefit for the functionality?
+
+* Is there an adequate mechanism to discover, deploy, express a
+  dependency on, and upgrade the functionality if implemented using an
+  extension mechanism? Are there consistent notions of releases, maturity,
+  quality, version skew, conformance, etc. for extensions?
+
+* Is it needed as a reference implementation exercising extension
+  points or other APIs?
+
+* Is the functionality sufficiently general-purpose?
+
+* Is it an area where we want to provide an opinionated solution
+  and/or where fragmentation would be problematic for users, or are
+  there many reasonable alternative approaches and solutions to the
+  problem?
+
+* Is it an area where we want to foster exploration and innovation in
+  the ecosystem?
+
+* Has the ecosystem produced adequate solutions on its own? For
+  instance, have ecosystem projects taken on requirements of the
+  Kubernetes project, if needed? Example: etcd3 added a number of features
+  and other improvements to benefit Kubernetes, so the project didn't
+  need to launch a separate storage effort.
+
+* Is there an acceptable home for the recommended ecosystem solution(s)?
+  Example: the [CNCF Sandbox](https://github.com/cncf/toc/blob/master/process/sandbox.md) is one possible home
+
+* Has the functionality been provided by the project/release/component
+  historically?
+
+## Technical scope details and rationale
```

Comment:
It's worth noting that most of the sections do not yet include a rationale, and therefore come across as simply a statement of decree, i.e. these things are arbitrarily declared to be in scope, with no reasons given.  It would be good to have a clear set of guiding principles for inclusion or exclusion (and, for example, cross references to numbered or named principles for each item).  In the absence of these, it would be fairly easy to make a counter-argument to many of the inclusions/exclusions, and end up in the kind of arguments that I think this document is attempting to circumvent.

And separately, there is a fair amount of stuff that is declared "grandfathered in", that superficially seems fairly arbitrary.  I think it would be useful to be clear about what's:

1. currently incorrectly treated as being in scope simply because of historical habits (and we haven't yet gotten around to kicking it out of scope), vs 
2. stuff we legitimately cant kick out of scope any more, ever, because that would somehow break backward compatibility, vs
3. stuff that is not truly in scope, but for some, possibly pragmatic reason, we still care about it.

I think lumping all three of the above together under the label "grandfathered in" creates some confusion and frustration.

Reply:
I few brain-dump principles that might apply:

1. "NICE-TO-HAVES" If it's fairy easy to build and run a useful cluster containing useful production cloud-native applications without this, then it's probably out of scope (e.g. istio, helm).
2. "NOT-REALLY-OPTIONAL-NATIVES" If >80?% of users use this thing with their clusters anyway, and it's not much use anywhere else than with kubernetes (to exclude things like etcd, which seems self-evidently out of scope) then it's probably in-scope.
3. "GENERAL-TOOLS" If this thing is one of a general class of stuff (e.g. CI or testing frameworks) that is or could be non-kubernetes-specific, but we just happen to have decided to build ourselves (e.g. prow?) then it's probably out of scope.
...
I can dump some more in a more useful place than a PR comment if you think that would be useful?

Reply:
Ack on "grandfathered" breakdown, though I'd say we need pragmatic reasons to "kick out" existing subprojects. Some past material for consideration includes:
https://github.com/kubernetes/community/blob/master/github-management/kubernetes-repositories.md
https://github.com/kubernetes/community/blob/master/incubator.md

Ack on the need for categories, also. There's some overlap with the earlier layers notion that's worth thinking about.